client/lib/localforage/index.js

```diff
@@ -16,11 +21,15 @@ const config = {
 	]
 };
 
+let _ready = false;
 // Promise that resolves when our localforage configuration has been applied
-const localForagePromise = new Promise( ( resolve ) => {
-	localforage.config( config );
-	resolve( localforage );
-} );
+const localForagePromise = localforage.defineDriver( localforageBypass )
+	.then( () => {
+		localforage.config( config );
```

Comment:
What does the timing look like if we check if we're in switch user mode here? Unfortunately I'm still seeing the sync-handler write to indexedDB while switched.